qlib/Mapper.qm

```diff
@@ -930,13 +931,42 @@ Mapper mapv(DataMap);
         }
 
         #! maps a single field to the target
-        private nothing mapFieldIntern(reference h, string key, hash rec, *int do_list) {
+        /**
+          * Performs the actual mapping
+          *
+          * @param h the hash to be updated with the mapped key/value pair
+          * @param key the column name (hash key) to be mapped (target field)
+          * @param rec input record - either single record of hash of lists (bulk);
+          *                           to increase performance, the input type
+          *                           (single record vs. bulk) is determined by
+          *                           the @ref do_list parameter. In case of
+          *                           bulk input, all the lists are supposed to
+          *                           have the same length @ref list_size
+          * @param do_list - whether the input record @ref rec is single record or
+          *                  bulk format (hash of lists)
+          * @param list_size - size of the lists in case the input is in bulk
+          *                    format (all lists must have the same length,
+          *                    asserted inside the function).
+          * @param crec - optional hash to be added to each processed record
```

Comment:
the `rec` hash already can be handled with a mix of "hash of lists" values and single values; why include an extra optional argument to this method when the data can be given in `rec` directly (i.e. added once to `rec` before input instead of processing it separately for each field in the mapper)?